students/WesleyM/Lesson9/mailroom.py

```diff
@@ -0,0 +1,120 @@
+class donor:
+    def __init__(self, name, donation = None):
+        if not donation:
+            donation = []
+        self.name = name
+        self.donation = donation
+        self.total = sum(self.donation)
+        if donation:
+            self.average = self.total/len(self.donation)
+        else:
+            self.average = 0
+    def add_donation(self, amount):
+        self.donation.append(amount)
+        self.total += amount
+        self.average = self.total / len(self.donation)
+
+class DonorDatabase:
+
+    def __init__(self, name):
+        if not name:
+            self.donors = []
+        else:
+            self.donors = name
+    
+    def add_donor(self, name):
+        self.donors.append(name)
+        
+    def get_donor(self, name):
+        for d in self.donors:
+            if d.name == name:
+                return d
+
+    def get_all_donor_names(self):
+        return [donor.name for donor in self.donors]
+        
+    def sort_donors(self):
+        return sorted(self.get_all_donor_names())
+    
+    def donor_input(self):
+        return input("Enter a donor name or input 'List' for a list of donors\n>")
+    
+    def send_thankyou(self):
```

Comment:
Ideally, you would separate all the data handling code from the code that deals with user input, so rather than make it a `DonorDatabase` method, I'd move this function to the module level